ddtrace/utils/version.py

```diff
@@ -0,0 +1,47 @@
+import typing
+
+import packaging.version
+
+
+def parse_version(version):
+    # type: (str) -> typing.Tuple[int, int, int]
+    """Convert a version string to a tuple of (major, minor, micro)
+
+    Examples::
+
+       1.2.3       -> (1, 2, 3)
+       1.2         -> (1, 2, 0)
+       1           -> (1, 0, 0)
+       1.0.0-beta1 -> (1, 0, 0)
+       2020.6.19   -> (2020, 6, 19)
+    """
+    # If we have any spaces/extra text, grab the first part
+    #   "1.0.0 beta1" -> "1.0.0"
+    #   "1.0.0" -> "1.0.0"
+    if " " in version:
+        version = version.split()[0]
+
+    # DEV: Do not use `packaging.version.parse`, we do not want a LegacyVersion here
+    # This will raise an InvalidVersion exception if it cannot be parsed
+    _version = packaging.version.Version(version)
+    return (
+        # Version.release was added in 17.1
+        # packaging >= 20.0 has `Version.{major,minor,micro}`, use the following
+        # to support older versions of the library
+        # https://github.com/pypa/packaging/blob/47d40f640fddb7c97b01315419b6a1421d2dedbb/packaging/version.py#L404-L417
+        _version.release[0] if len(_version.release) >= 1 else 0,
+        _version.release[1] if len(_version.release) >= 2 else 0,
+        _version.release[2] if len(_version.release) >= 3 else 0,
+    )
+
+
+def parse_version_safe(version):
```

Comment:
I just added this to account for the update in `pprof.pyx` where we were looking to ensure that the resulting parsed version was a `Version` and not a `Legacy` version.

Instead of assume we want to raise in that case, I added this instead. Worth discussing the right approach

Reply:
Isn't this logically the same as using `Version` and `Legacy` version in our code ([legacy versions will always be less than a version](https://packaging.pypa.io/en/latest/version.html#packaging.version.LegacyVersion))? Why not just use the `packaging` datastructures?

Reply:
Yeah, see https://github.com/DataDog/dd-trace-py/pull/2395/files#r628142671

think you are right, there is a better approach